src/main/java/testapp/adapter/memory/InMemoryStudentRepository.java

```diff
@@ -0,0 +1,38 @@
+package testapp.adapter.memory;
+
+import testapp.core.Student;
+import testapp.core.StudentAlreadyExistsException;
+import testapp.core.StudentRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class InMemoryStudentRepository implements StudentRepository {
```

Comment:
I would expect in a following PR to have a datastore implementation of the repository